#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t

gen = ParameterGenerator()

gen.add("outline_count", int_t, 0, "Outline count to mow before filling", 3, 0, 255)
gen.add("outline_overlap_count", int_t, 0, "Outlines to overlap with fill", 0, 0, 255)
gen.add("outline_offset", double_t, 0, "Additional outer outline offset, use positive values for safety, negative values to enlarge the area", 0.0, -1.0, 1.0)
gen.add("mow_angle_offset", double_t, 0, "Mowing angle offset", 0, -180, 180)
gen.add("mow_angle_offset_is_absolute", bool_t, 0, "If true then the offset is relative to East, if false then the offset is relative to the auto-detected angle.", False)
gen.add("mow_angle_increment", double_t, 0, "Mowing angle automatic increment. Will be added to the offset every time the entire map is finished", 0, 0, 180)
gen.add("tool_width", double_t, 0, "Width of the mower", 0.14, 0.1, 2)

exit(gen.generate("mower_logic", "tasklist", "Tasklist"))
